apps/Standalone/src/mcp/app/McpStandard.tsx

```diff
@@ -70,66 +86,78 @@ const AwaitingConnection = () => {
     error: workflowError,
-  } = useWorkflowApp(isValidResourceId ? appId : '', hostingPlan, isValidResourceId);
+  } = useWorkflowApp(isValidResourceId ? appId : '', hostingPlan, isValidResourceId && !!appId);
 
   const canonicalLocation = useMemo(
     () => WorkflowUtility.convertToCanonicalFormat(workflowAppData?.location ?? 'westus'),
     [workflowAppData]
   );
 
-  const resourceDetails = useMemo(() => {
-    if (!isValidResourceId) {
-      return null;
-    }
-    return new ArmParser(appId);
-  }, [appId, isValidResourceId]);
-
   const isConnected = !!(isValidResourceId && workflowAppData && resourceDetails && !workflowError && !isWorkflowLoading);
 
-  const services = useMemo(() => {
-    const siteResourceId = new ArmParser(appId ?? '').topmostResourceId;
-    const armUrl = 'https://management.azure.com';
-    const baseUrl = `${armUrl}${siteResourceId}/hostruntime/runtime/webhooks/workflow/api/management`;
-    const defaultServiceParams = { baseUrl, httpClient, apiVersion };
-
-    const searchService = new StandardSearchService({
-      ...defaultServiceParams,
-      apiHubServiceDetails: {
-        apiVersion: '2018-07-01-preview',
-        subscriptionId: resourceDetails?.subscriptionId ?? '',
-        location: canonicalLocation,
-      },
-      isDev: false,
-      unsupportedConnectorIds: ['/subscriptions/#subscription#/providers/Microsoft.Web/locations/#location#/managedApis/gmail'],
-    });
-
-    const hostService: IHostService = {
-      fetchAndDisplayContent: (title: string, url: string, type: ContentType) => console.log(title, url, type),
-      openWorkflowParametersBlade: () => console.log('openWorkflowParametersBlade'),
-      openConnectionResource: (connectionId: string) => console.log('openConnectionResource:', connectionId),
-      openMonitorView: (workflowName: string, runName: string) => console.log('openMonitorView:', workflowName, runName),
-    };
-
-    return {
-      searchService,
-      hostService,
-    };
-  }, [appId, canonicalLocation, resourceDetails]);
+  const { data: tenantId } = useCurrentTenantId();
+  const { data: objectId } = useCurrentObjectId();
+
+  const services = useMemo(
+    () => getServices(appId as string, workflowAppData as WorkflowApp, tenantId, objectId, canonicalLocation),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [workflowAppData, tenantId, canonicalLocation, appId]
+  );
+
+  useEffect(() => {
+    if (shouldReload) {
+      const logicAppId = getWorkflowAppIdFromStore();
+      if (equals(logicAppId, workflowAppData?.id)) {
+        mcpStore.dispatch(
+          resetMcpStateOnResourceChange(
+            getResourceBasedServices(logicAppId as string, workflowAppData as WorkflowApp, tenantId, objectId, canonicalLocation)
+          )
+        );
+        setShouldReload(false);
+      }
+    }
+  }, [canonicalLocation, objectId, shouldReload, tenantId, workflowAppData]);
+
+  const onResourceChange = useCallback(async () => {
```

Comment:
Confused here why we're using multiple appIds (logicappId and appId) is it that we currently have it in devtoolbox and when we move it out we want to pull it from store?

Reply:
yes... I am waiting for this control to be present in wizard so this can be re-initialized properly